holidays/countries/christmas_islands.py

```diff
@@ -0,0 +1,176 @@
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC
+from holidays.groups import (
+    ChristianHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+    StaticHolidays,
+    ChineseCalendarHolidays,
+)
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    MON_TO_NEXT_TUE,
+    SAT_SUN_TO_NEXT_MON,
+    SAT_SUN_TO_NEXT_MON_TUE,
+)
+
+
+class ChristmasIslands(
+    ObservedHolidayBase,
+    ChristianHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+    StaticHolidays,
+    ChineseCalendarHolidays,
+):
+    """
+    Christmas Islands Holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Christmas_Island>
+        * <https://web.archive.org/web/20240519034837/https://www.infrastructure.gov.au/sites/default/files/documents/a11-2023-2024-public-holidays-christmas-island.pdf>
+        * <https://web.archive.org/web/20250413083314/https://www.infrastructure.gov.au/sites/default/files/documents/a21-2024-administrator-community-bulletin-cki-public-holidays-2025.pdf>
+    """
+
+    country = "CX"
+    default_language = "en_CX"
+    # %s (observed).
+    observed_label = tr("%s (observed)")
+    # %s (estimated).
+    estimated_label = tr("%s (estimated)")
+    # %s (observed, estimated).
+    observed_estimated_label = tr("%s (observed, estimated)")
+    supported_languages = ("en_CX", "en_US")
+    start_year = 1985
+
+    def __init__(self, *args, islamic_show_estimated: bool = True, **kwargs):
+        ChineseCalendarHolidays.__init__(self)
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=ChristmasIslandsIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        StaticHolidays.__init__(self, ChristmasIslandsStaticHolidays)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day(tr("New Year's Day")))
+
+        # Australia Day.
+        self._add_observed(self._add_holiday_jan_26(tr("Australia Day")))
+
+        # Chinese New Year.
+        name = tr("Chinese New Year")
+        self._add_observed(self._add_chinese_new_years_day(name))
+        self._add_observed(self._add_chinese_new_years_day_two(name))
+
+        # Labour Day.
+        self._add_holiday_4th_mon_of_mar(tr("Labour Day"))
```

Comment:
AFAIK only these 4 years uses the 4th Monday of March instead:
```suggestion
        # Labor Day.
        name = tr("Labour Day")
        if self._year in (2009, 2010, 2014, 2021):
            self._add_holiday_4th_mon_of_mar(name)
        else:
            self._add_holiday_3rd_mon_of_mar(name)
```